etna/models/nn/nbeats/utils.py

```diff
@@ -0,0 +1,105 @@
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import Optional
+
+import numpy as np
+
+from etna import SETTINGS
+
+if SETTINGS.torch_required:
+    import torch
+
+
+def _create_or_update(param: Optional[Dict], name: str, value: Any):
+    """Create new parameters dict or add field to existing one."""
+    if param is None:
+        param = {name: value}
+    else:
+        param[name] = value
+    return param
+
+
+def default_torch_device() -> "torch.device":
+    """Return CUDA device if available."""
+    return torch.device("cuda" if torch.cuda.is_available() else "cpu")
+
+
+def to_tensor(x: Any) -> "torch.Tensor":
+    """Convert data to tensor and put on default device.
+
+    Parameters
+    ----------
+    x:
+        Initial data for the tensor. Can be a list, tuple, NumPy ndarray, scalar, and other types.
+
+    Returns
+    -------
+    :
+        Input data as tensor.
+    """
+    return torch.tensor(x, dtype=torch.float32)
+
+
+def prepare_train_batch(
```

Comment:
Is this ok that collate function does this logic? I thought that collate function is only responsible for grouping batches into one object.